### ISSUE TITLE
Perception: L4-Clustering for Segmentation E5 - Example doesn't work on native system

### ISSUE THREAD

**Issue by Shreeyak** [udacity/robotics-beta-test-issues#590]
I copied the k-means example provided and ran it on my local system. It generated an error aat line 53, in func cv2.kmeans(). According to the documentation that I found, an extra parameter is being passed, which generates an error.
compactness, label, center = cv2.kmeans(data, k_clusters, None, criteria, 10, cv2.KMEANS_RANDOM_CENTERS)

should be:
compactness, label, center = cv2.kmeans(data, k_clusters, criteria, 10, cv2.KMEANS_RANDOM_CENTERS)

Not sure why this change occurs. Which version of OpenCV is used in the example?

Also, when I do run the script, it does not show the plot. Or it closes so quickly that I can't see it. Do add suggestions for the same in the material.

At this page: https://classroom.udacity.com/nanodegrees/nd209-beta/parts/571766f0-9631-484d-a69f-94ea58110725/modules/e5bfcfbd-3f7d-43fe-8248-0c65d910345a/lessons/2cc29bbd-5c51-4c3e-b238-1282e4f24f42/concepts/d0f186bc-ad58-4cca-8013-2434ae9802a2

**Comment by Shreeyak** [comment]
Okay, so I mucked around a lot with pip and jupyter (some mixup with pip not upgrading, then pointing to python3, etc). Now I got pip pointing to python2 and errors sorted out.

In this env (running python2), Running the code in Jupyter gives me an error in cv2.kmeans() and I must add the parameter `None` as in the lesson. Even executing it from terminal using command `python <filename>` needs extra parameter as in lesson. I'm sure that python pointed to version 2.7 earlier, as now, but don't know why it did not give error earlier.

That said,  it still doesn't give me a plot, even in Jupyter. any suggestions?
